mlir/lib/Dialect/Rock/Generator/Conv2dGenerator.cpp

```diff
@@ -875,14 +883,41 @@ LogicalResult Conv2dGenerator::genConvModule(ModuleOp &module, int kernel_id,
     block->push_front(convOp);
   } break;
   case ConvOpType::BwdData: {
-    auto convOp = builder.create<Conv2DBwdDataOp>(
-        builder.getUnknownLoc(), ArrayRef<Type>{}, args, attributes);
-    block->push_front(convOp);
+    if (kernelId < 0) {
```

Comment:
Can this happen? I thought around line 766 when rawKernelId comes in, it can no longer be less than 0?

Reply:
Okay gotcha, I misread what `rawKernelId` is and turns out this is not an id but rather an index to reference from `kernelIds` (in backward data case only). This is somewhat confusing.